src/mqt/ddsim/qasmsimulator.py

```diff
@@ -73,22 +74,57 @@ def target(self):
     def max_circuits(self):
         return None
 
-    def run(self, quantum_circuits: QuantumCircuit | list[QuantumCircuit], **options: dict[str, Any]) -> DDSIMJob:
+    @staticmethod
+    def _bind_parameters(
+        quantum_circuits: list[QuantumCircuit], parameter_values: Sequence[Sequence[float]] | None = None
```

Comment:
If you make the `parameter_values` a `Sequence` you might as well make the `quantum_circuits` parameter a `Sequence`.
Technically, you could go even one step further and make both an `Iterable`, which is one step more generic than `Sequence` and also allows, e.g., to pass `dict[str, QuantumCircuit]`. In general, typing information for parameters in functions should be as generic as possible, while the return type should be as precise as possible.

Reply:
Changing the types to Sequence should work. However, Iterable is too general since in many parts of the code we are using indexation, which doesn't work so straightforward for dictionaries.

Reply:
Alright, makes sense. I believe the change to `Sequence` is still worth it though 👍🏼